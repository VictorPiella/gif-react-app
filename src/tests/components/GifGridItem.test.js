import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';



describe('Test  <GifGridItem />', () => {

    const title = "A title";

    const url = "https://localhost/text.jpg";

    const wrapper = shallow( <GifGridItem title={ title } url={ url } /> );

    test('show the component correctly', () => {

        expect ( wrapper ).toMatchSnapshot();
      
    })

    test('should had a <p> with the title', () => {

        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );

    })

    test('should has image = url and alt = props', () => {

        const img = wrapper.find('img');

        expect ( img.prop('src') ).toBe(url);
        expect ( img.prop('alt') ).toBe(title);
        
    })

    test('should had clÃ±ass fade in', () => {

        const div = wrapper.find('div');

        expect ( div.prop('className').includes('animate__fadeIn') ).toBe( true )
        
    })
    
    
})