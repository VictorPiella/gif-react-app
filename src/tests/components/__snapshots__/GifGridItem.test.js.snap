// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test  <GifGridItem /> show the component correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="A title"
    src="https://localhost/text.jpg"
  />
  <p>
     
    A title
     
  </p>
</div>
`;
